components/monitoring/base/prometheus-servicemonitors.yaml

```diff
@@ -0,0 +1,46 @@
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: prometheus
+  labels:
+    prometheus: appstudio-kcp-workload
+spec:
+  endpoints:
+  - bearerTokenSecret:
+      key: ""
+    interval: 15s
+    port: oauth2-proxy
+    scheme: HTTPS
+    tlsConfig:
+      caFile: /etc/prometheus/tls/tls.crt # volume mounted in the `prometheus` container of the prometheus pods
+      serverName: "prometheus-oauth2.appstudio-workload-monitoring.svc"
+  namespaceSelector:
+    matchNames:
+    - appstudio-workload-monitoring
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: prometheus
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: spi-operator
+  labels:
+    prometheus: appstudio-kcp-workload
+spec:
+  endpoints:
+    - bearerTokenSecret:
+        name: prometheus-k8s-token-x6qjc
+        key: token
+      scheme: https
+      tlsConfig:
+        insecureSkipVerify: true
+      port: https
+  namespaceSelector: {}
```

Comment:
could we label the `appstudio` namespaces to avoid scanning _all_ namespaces?

Reply:
we want to scan all the namespaces here. Because we dont know about the generated namespace name by ArgoCD.

Reply:
I know we can't control the name of the generated namespaces, so could we add labels on these namespaces?

Reply:
thats a great idea @MatousJobanek any thoughts on this please?

Reply:
I love the idea of adding an appstudio label to all of "our" namespaces, but I don't know how to do it.

Reply:
as of now, (in kcp environment) there is no way to add some labels to `kcp-*` namespaces. 
Instaed of using ServiceMonitors, we should use PodMonitors because we can control labels at Pods

Reply:
> Instead of using ServiceMonitors, we should use PodMonitors because we can control labels at Pods

how does it defer from labels on services?

Reply:
let me look into how prometheus supports the podmonitor. then also the issue remains same we need to grep through all the namespaces in the cluster.